R/LongitudinalQuantities.R

```diff
@@ -0,0 +1,214 @@
+
+#' @include DataJoint.R
+#' @include Quantities.R
+NULL
+
+
+#' `LongitudinalQuantities` Object & Constructor Function
+#'
+#' Constructor function to generate a `LongitudinalQuantities` object.
+#'
+#' @param object ([`JointModelSamples`]) \cr Samples as drawn from a Joint Model
+#'
+#' @param groups (`character`, `NULL`)\cr which patients to calculate the desired
+#' quantities for.
+#'
+#' @param time_grid (`numeric` or `NULL`)\cr a vector of time points to calculate the desired
+#' quantity at. If `NULL` will be set to `seq(0, max_longitudinal_time, length = 201)`
+#'
+#' @details
+#' Note that unlike [SurvivalQuantities], [LongitudinalQuantities] does not support
+#' group aggregation.
+#'
+#' @slot quantities (`Quantities`)\cr The sampled quantities. Should contain 1 element per
+#' element of `group`
+#' @slot groups (`list`)\cr See argument section for details
+#' @slot time_grid (`numeric`)\cr See argument section for details
+#' @slot data ([`DataJoint`])\cr The data that the Joint Model was fitted to to produce
+#' the samples/quantities
+#'
+#'
+#' @family LongitudinalQuantities
+#' @name LongitudinalQuantities-class
+#' @export LongitudinalQuantities
+.LongitudinalQuantities <- setClass(
+    "LongitudinalQuantities",
+    slots = c(
+        "quantities" = "Quantities",
+        "data" = "DataJoint",
+        "groups" = "list",
+        "time_grid" = "numeric"
+    )
+)
+
+#' @rdname LongitudinalQuantities-class
```

Comment:
would maybe be more natural to put the roxygen `@param` etc. here? It will end up in the same page but for developers it is easier if it is directly above the thing it documents